### PR TITLE
Tags Implementation

### PR DESCRIPTION
**Pull request by minikin**
- Add Tags Implementation

**NOTE**: It looks like that we don't have an implementation point for tags yet.

┆Issue is synchronized with this [Jira Task](https://highvibe.atlassian.net/browse/HVN1-177) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

**Comment by minikin** [comment]
@JasonHighVibe I'll merge this for now. I'll implement a full audio upload/edit workflow for mobile after refactoring.

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 1]
```diff
@@ -61,6 +51,51 @@ class _UploadAudioPageState
     );
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    when(
```
Comment (AlexVegner) [inline lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:57]: Mobx autorun, reaction, when should be properly disposed.
https://pub.dev/packages/mobx#reactions

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 2]
```diff
@@ -61,6 +51,51 @@ class _UploadAudioPageState
...
+        }
+
+        if (isAudioFileType) {
+          controller.nextStep(data: file);
+        } else {
+          Modular.to.pop();
+        }
+      },
+    );
+    when((_) => controller.currentStep == UploadAudioStep.success, () async {
```
Comment (AlexVegner) [inline lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:78]: Mobx autorun, reaction, when should be properly disposed.
https://pub.dev/packages/mobx#reactions

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 3]
```diff
@@ -61,6 +51,51 @@ class _UploadAudioPageState
...
+//      await ProfileModule.toEditAudio(controller.audio);
```
Comment (AlexVegner) [inline lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:79]: Minor issue: Remove commented code.

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 4]
```diff
@@ -61,6 +51,51 @@ class _UploadAudioPageState
...
+        title: 'Karma Reward',
+        subTitle: 'You received ${controller.audio.karmaReward} Karma',
+      );
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    animationController = AnimationController(vsync: this)
```
Comment (AlexVegner) [inline lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:92]: Override dispose method and dispose the AnimationController instance.
```dart
@override
dispose() {
  animationController.dispose(); // you need this
  super.dispose();
}
```

File: lib/widgets/tags/tag_item.dart [hunk 5]
```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:highvibe/widgets/tags/tag_item_remove_button_view_model.dart';
+
+typedef OnLongPressedCallback = void Function();
```
Comment (AlexVegner) [inline lib/widgets/tags/tag_item.dart:6]: Minor: Use dart`dart VoidCalback` for `dart void Function();`
```
/// Signature of callbacks that have no arguments and return no data.
typedef VoidCallback = void Function();
```

File: lib/widgets/tags/tag_item.dart [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+typedef OnPressedCallback = void Function();
```
Comment (AlexVegner) [inline lib/widgets/tags/tag_item.dart:7]: Minor: Use dart`dart VoidCalback` for `dart void Function();`
```
/// Signature of callbacks that have no arguments and return no data.
typedef VoidCallback = void Function();
```

File: lib/widgets/tags/tag_item_remove_button_view_model.dart [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+
+typedef OnDeletedCallback = void Function();
```
Comment (AlexVegner) [inline lib/widgets/tags/tag_item_remove_button_view_model.dart:3]: Minor: Use dart`dart VoidCalback` for `dart void Function();`
```
/// Signature of callbacks that have no arguments and return no data.
typedef VoidCallback = void Function();
```

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart (hunk 1 above)
Comment (minikin) [reply lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:57]: That's not part of this PR. Not going to fix it.

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart (hunk 2 above)
Comment (minikin) [reply lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:78]: That's not part of this PR. Not going to fix it.

File: lib/modules/profile/current_user/upload_audio/upload_audio_page.dart (hunk 4 above)
Comment (minikin) [reply lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:92]: That's not part of this PR. Not going to fix it.

File: lib/widgets/tags/tag_item.dart (hunk 6 above)
Comment (minikin) [reply lib/widgets/tags/tag_item.dart:7]: We need to have a clear distinction between actions so `OnLongPressedCallback` and `OnPressedCallback` make more sense here. Use of `VoidCalback ` doesn't add any value or improvements to the code base.

File: lib/widgets/tags/tag_item_remove_button_view_model.dart (hunk 7 above)
Comment (minikin) [reply lib/widgets/tags/tag_item_remove_button_view_model.dart:3]: the same as a previous one :)

**Pull request merged**